src/provider/provider_tracking.c

```diff
@@ -429,6 +466,132 @@ static const char *trackingName(void *provider) {
     return umfMemoryProviderGetName(p->hUpstream);
 }
 
+static umf_result_t trackingGetIpcHandleSize(void *provider, size_t *size) {
+    umf_tracking_memory_provider_t *p =
+        (umf_tracking_memory_provider_t *)provider;
+    return umfMemoryProviderGetIPCHandleSize(p->hUpstream, size);
+}
+
+static umf_result_t trackingGetIpcHandle(void *provider, const void *ptr,
+                                         size_t size, void *ipcData) {
+    umf_tracking_memory_provider_t *p =
+        (umf_tracking_memory_provider_t *)provider;
+    umf_result_t ret = UMF_RESULT_SUCCESS;
+    size_t ipcDataSize = 0;
+    int cached = 0;
+    do {
+        void *value = critnib_get(p->ipcCache, (uintptr_t)ptr);
+        if (value) { //cache hit
+            ipc_cache_value_t *cache_value = (ipc_cache_value_t *)value;
+            memcpy(ipcData, cache_value->ipcData, cache_value->size);
+            cached = 1;
+        } else {
+            ret =
+                umfMemoryProviderGetIPCHandle(p->hUpstream, ptr, size, ipcData);
```

Comment:
> I tried but I don't get what each 'size' means :( can you change the names of these variables so they become more meaningful? or add some comments?

I have renamed `ipc_cache_value_t::size` to `ipc_cache_value_t::ipcDataSize`. And also I added comments to `ipc_cache_value_t` data structure and `umf_ipc_data_t` data structure. Hope it will make things clear.

> Also, as I saw Level Zero doesn't use this arg, right?

Correct. But the same story with `ze_memory_provider_free` function - there `bytes` parameter is not used in case of L0, but in general case it might be used by other implementation that is why we have the size parameter in these two interfaces of memory provider.